libsplinter/src/network/auth/handlers/v1_handlers.rs

```diff
@@ -0,0 +1,456 @@
+                    protocol_request.auth_protocol_max,
+                );
+
+                // Send error message if version is not agreed upon
+                if version == 0 {
+                    let response = AuthorizationMessage::AuthorizationError(
+                        AuthorizationError::AuthorizationRejected(
+                            "Unable to agree on protocol version".into(),
+                        ),
+                    );
+
+                    let mut msg = NetworkMessage::new();
+                    msg.set_message_type(NetworkMessageType::AUTHORIZATION);
+                    msg.set_payload(
+                        IntoBytes::<authorization::AuthorizationMessage>::into_bytes(response)?,
+                    );
+
+                    sender
+                        .send(context.source_id().clone(), msg.write_to_bytes()?)
+                        .map_err(|(recipient, payload)| {
+                            DispatchError::NetworkSendError((recipient.into(), payload))
+                        })?;
+
+                    if self
+                        .auth_manager
+                        .next_remote_state(
+                            context.source_connection_id(),
+                            AuthorizationAction::Unauthorizing,
+                        )
+                        .is_err()
+                    {
+                        warn!(
+                            "Unable to update state to Unauthorizing for {}",
+                            context.source_connection_id(),
+                        )
+                    };
+
+                    return Ok(());
+                };
+
+                debug!(
+                    "Sending agreed upon protocol version: {} and authorization types",
+                    version
+                );
+
+                let response = AuthorizationMessage::AuthProtocolResponse(AuthProtocolResponse {
+                    auth_protocol: version,
+                    accepted_authorization_type: vec![PeerAuthorizationType::Trust],
+                });
+
+                let mut msg = NetworkMessage::new();
+                msg.set_message_type(NetworkMessageType::AUTHORIZATION);
+                msg.set_payload(
+                    IntoBytes::<authorization::AuthorizationMessage>::into_bytes(response)?,
+                );
+
+                sender
+                    .send(context.source_id().clone(), msg.write_to_bytes()?)
+                    .map_err(|(recipient, payload)| {
+                        DispatchError::NetworkSendError((recipient.into(), payload))
+                    })?;
+            }
+            Ok(next_state) => panic!("Should not have been able to transition to {}", next_state),
+        }
+        Ok(())
+    }
+}
+
+/// Return the supported protocol version that matches the min/max provided. If there is no
+/// matching protocol version 0 is returned.
+fn supported_protocol_version(min: u32, max: u32) -> u32 {
```

Comment:
Should this return a Result instead of an int with a magic value?

Reply:
Probably, but I think this matches how these protocol version matches are checked elsewhere in the code base.  We should add a story to make that sort of conversion, but it shouldn't block this PR

Reply:
I'll add a story.